spec/services/users/username_generator_spec.rb

```diff
@@ -0,0 +1,31 @@
+require "rails_helper"
+
+RSpec.describe Users::UsernameGenerator, type: :service do
+  let(:user) { create(:user) }
```

Comment:
Nitpick: I believe this does not _need_ to be a persisted user?

```suggestion
  let(:user) { build(:user) }
```